.github/workflows/docker-build-and-push.yaml

```diff
@@ -0,0 +1,48 @@
+name: Docker Build and Push
+
+on:
+  push:
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+
+    env:
+      IMAGE: matejak/argbash
+
+    steps:
+      - uses: actions/checkout@v2
+      
+      - name: Set tags
+        id: set_tags
+        shell: bash
+        run: |
+          if [[ "$GITHUB_REF" = refs/tags/* ]]; then
+            version="${GITHUB_REF#refs/tags/}"
+            tags="latest,$version"
+          else
+            branch="${GITHUB_REF#refs/heads/}"
+            branch="branch-${branch/\//-}"
+            tags="$branch"
+          fi
+          echo "::set-output name=tags::$tags"
+          echo "The tags were set to '$tags'"
+          
+      - uses: docker/build-push-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+          repository: ${{ env.IMAGE }}
+          dockerfile: docker/Dockerfile
+          add_git_labels: true
+          tags: ${{ steps.set_tags.outputs.tags }}
+          tag_with_sha: true
+
+      - name: Update Docker Hub Description
+        if: startsWith(github.ref, 'refs/tags/')
+        uses: peter-evans/dockerhub-description@v2
+        env:
+          DOCKERHUB_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+          DOCKERHUB_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+          DOCKERHUB_REPOSITORY: ${{ env.IMAGE }}
+          README_FILEPATH: docker/README.md
```

Comment:
> That's a great addition, however I have one question:
> 
> If you take a look at the Docker Hub page, you will notice that the description there is different, centred around the fact that it is an image, which has to be used differently than a command-line app. Would it be possible to somehow keep those descriptions separate?

@matejak I already treated this case in this piece of code. It updates the Docker Hub description with the `docker/README.md` file every time we push the image as latest. This way we also automate the update of the Docker Hub description by reading it from the GitHub directly.

Reply:
But I'm thinking now, it would be also useful to update the Docker Hub description every time the `docker/README.md` gets updated, right?

Reply:
I agree, let's do it every commit.

Reply:
Did it.